.github/workflows/scripts/assign-milestone-to-merged-pr.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Script to automatically assign a milestone to a pull request when it's merged.
+ *
+ * @package WooCommerce/GithubActions
+ */
+
+// phpcs:disable WordPress.Security.EscapeOutput.OutputNotEscaped, WordPress.WP.AlternativeFunctions
+
+global $repo_owner, $repo_name, $github_token, $graphql_api_url;
+
+/**
+ * Grab/process input.
+ */
+
+$repo_parts = explode( '/', getenv( 'GITHUB_REPOSITORY' ) );
+$repo_owner = $repo_parts[0];
+$repo_name  = $repo_parts[1];
+
+$pr_id           = getenv( 'PULL_REQUEST_ID' );
+$github_token    = getenv( 'GITHUB_TOKEN' );
+$graphql_api_url = getenv( 'GITHUB_GRAPHQL_URL' );
+
+/**
+ * Select the milestone to be added:
+ *
+ * 1. Get the first 10 milestones sorted by creation date descending.
+ *    (we'll never have more than 2 or 3 active milestones but let's get 10 to be sure).
+ * 2. Discard those not open or whose title is not a proper version number ("X.Y.Z").
+ * 3. Sort descending using version_compare.
+ * 4. Get the oldest one that does not have a corresponding "release/X.Y" branch.
+ */
+
+echo "Getting the list of milestones...\n";
+
+$query      = "
+  repository(owner:\"$repo_owner\", name:\"$repo_name\") {
+    milestones(first: 10, orderBy: {field: CREATED_AT, direction: DESC}) {
```

Comment:
```suggestion
    milestones(first: 10, states: [OPEN], orderBy: {field: CREATED_AT, direction: DESC}) {
```

Let's add states to this query here itself instead of verifying in a loop

Reply:
Neat, I didn't know that the GH API allowed that.